torch/_inductor/codegen/triton.py

```diff
@@ -1639,13 +1647,22 @@ def call_kernel(self, code, name: str):
         grid = []
         # TODO(jansel): if there are constants, we shouldn't bother passing them as args
         for tree in self.range_trees:
+            assignment = False
             if isinstance(tree.numel, (sympy.Integer, sympy.Symbol)):
                 expr = pexpr(tree.numel)
             else:
+                assignment = True
                 expr = f"{name}_{tree.prefix}numel"
-                code.writeline(f"{expr} = {pexpr(tree.numel)}")
+                # TODO(voz): Tragic. This should at the very least be a util to slapp on declare and ending.
+                # The real fix here is to revisit our cross language calling convention.
+                code.writeline(
+                    f"{code.declare}{expr} = {pexpr(tree.numel)}{code.ending}"
```

Comment:
`pexpr` will break for more complicated expressions, it's python expression printer

Reply:
Yea, for sure, but I am just trying to keep the tests passing here. Ultimately we need to rewrite this to either split the entire first pass of aot into C++ (as you said) or to write these calling convention deliberately instead of just relying on it working by accident and then hitting these pitfalls.